Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -270,6 +270,101 @@ Alternatively to validation functions type conversion by using `Integer.parseInt
 * Drozer
 
 
+### Testing Custom URL Schemes
+
+#### Overview
+
+Both Android and iOS allow inter-app communication through the use of custom URL schemes. These custom URLs allow other applications to perform specific actions within the application hosting the custom URL scheme. Much like a standard web URL that might start with `https://`, custom URIs can begin with any scheme prefix and usually define an action to take within the application and parameters for that action.
+
+As a contrived example, consider: `sms://compose/to=your.boss@company.com&message=I%20QUIT!&sendImmediately=true`. Using something like this embedded as a link on a web page, when clicked by a victim on their mobile device, calling the custom URI with maliciously crafted parameters might trigger an SMS to be sent by the vulnerable SMS application with attacker defined content.
```

Comment:
Might want to add a sentence about impacts of such vulnerabilities, e.g., sending SMS to premium numbers (monetary), to predefined addresses (collecting phone numbers, rigging votes for talent shows).

The second sentence is very complex. Maybe it could be:

When a victim clicks such a link on a web page in their mobile browser, the vulnerable SMS application will send the SMS message with the maliciously crafted content.